lib/Alchemy/Phrasea/Controller/Prod/Property.php

```diff
@@ -64,49 +65,34 @@ public function displayStatusProperty(Application $app, Request $request)
         }
 
         $records = RecordsRequest::fromRequest($app, $request, false, ['chgstatus']);
-        $databoxStatus = \databox_status::getDisplayStatus($app);
-        $statusBit = $nRec = [];
 
-        foreach ($records as $record) {
-            //perform logic
-            $sbasId = $record->get_databox()->get_sbas_id();
-
-            if (!isset($nRec[$sbasId])) {
-                $nRec[$sbasId] = ['stories' => 0, 'records' => 0];
-            }
-
-            $nRec[$sbasId]['records']++;
-
-            if ($record->is_grouping()) {
-                $nRec[$sbasId]['stories']++;
-            }
+        if (count($records->databoxes()) > 1) {
+            return new Response($app['twig']->render('prod/actions/Property/index.html.twig', [
+                'records'   => $records
+            ]));
+        }
 
-            if (!isset($statusBit[$sbasId])) {
-                $statusBit[$sbasId] = isset($databoxStatus[$sbasId]) ? $databoxStatus[$sbasId] : [];
+        $databox = current($records->databoxes());
+        $statuses = $databox->getStatusStructure()->getStatuses();
 
-                foreach (array_keys($statusBit[$sbasId]) as $bit) {
-                    $statusBit[$sbasId][$bit]['nset'] = 0;
+        foreach ($records->received() as $record) {
+            $statuses = $statuses->map(function($status) use ($record) {
+                $statusSet = \databox_status::bitIsSet($record->getStatusBitMask(), $status['bit']);
+                if (!isset($status['flag'])) {
+                    $status['flag'] = (int) $statusSet;
```

Comment:
I'm not a big fan a exposing internal status structure (the `statuses` array), an manipulate it from outside. Maybe we can avoid that with a method on the record or somewhere else ?